core/src/main/java/org/sonarsource/sonarlint/core/serverapi/issue/IssueApi.java

```diff
@@ -63,9 +63,12 @@ public IssueApi(ServerApiHelper serverApiHelper) {
    *
    * @param key project key, or file key.
    */
-  public DownloadIssuesResult downloadVulnerabilitiesForRules(String key, Set<String> ruleKeys, ProgressWrapper progress) {
+  public DownloadIssuesResult downloadVulnerabilitiesForRules(String key, Set<String> ruleKeys, @Nullable String branchName, ProgressWrapper progress) {
     StringBuilder searchUrl = new StringBuilder();
     searchUrl.append(getVulnerabilitiesUrl(key, ruleKeys));
+    if (branchName != null) {
```

Comment:
Nitpick: the condition can be pushed to the builder method (lowers cognitive complexity).